botorch/models/approximate_gp.py

```diff
@@ -386,6 +372,85 @@ def __init__(
 
         self.to(train_X)
 
+    def init_inducing_points(
+            self,
+            inputs: Tensor,
+    ) -> Tensor:
+        r"""
+        Reinitialize the inducing point locations in-place with the current kernel
+        applied to `inputs`.
+        The variational distribution and variational strategy caches are reset.
+
+        Args:
+            inputs: input_data (*batch_shape, n, d)
+
+        Returns:
+            (*batch_shape, m, d) selected inducing point locations
+        """
+        var_strat = self.model.variational_strategy
+        clear_cache_hook(var_strat)
+        if hasattr(var_strat, 'base_variational_strategy'):
+            var_strat = var_strat.base_variational_strategy
+
+        with torch.no_grad():
+            num_inducing = var_strat.inducing_points.size(-2)
+            inducing_points = _select_inducing_points(
+                inputs, self.model.covar_module, num_inducing, self._input_batch_shape
+            )
+            var_strat.inducing_points.copy_(inducing_points)
+            var_strat.variational_params_initialized.fill_(0)
+
+        return inducing_points
+
+
+def _select_inducing_points(
+        inputs: Tensor,
+        covar_module: Module,
+        num_inducing: int,
+        input_batch_shape: torch.Size,
+) -> Tensor:
+    r"""
+    Utility function that evaluates a kernel at `inputs` and selects inducing point
+    locations based on the pivoted Cholesky heuristic.
+
+    Args:
+        inputs: input data (*batch_shape, n, d)
+        covar_module: GPyTorch Module returning a LazyTensor kernel matrix
+        num_inducing: the max number (m) of inducing points (m <= n)
+        input_batch_shape: non-task-related batch shape
+
+    Returns:
+        (*batch_shape, m, d) inducing point locations
+    """
+
+    train_train_kernel = covar_module(inputs).evaluate_kernel()
+    # base case
+    if train_train_kernel.ndimension() == 2:
+        inducing_points = _pivoted_cholesky_init(
+            inputs, train_train_kernel, max_length=num_inducing
+        )
+    # multi-task case
+    elif inputs.ndimension() == 2 and len(input_batch_shape) == 0:
+        inducing_points = _pivoted_cholesky_init(
+            inputs, train_train_kernel[0], max_length=num_inducing
+        )
+    # batched input cases
```

Comment:
I think this is a reasonable way to handle batched inputs for now. probably not worth investing too much time in, bc it will change once batched pivoted cholesky decompositions become available.

Reply:
@wjmaddox any updates on using gpytorch's pivoted Cholesky algo / making that support batch mode?

Reply:
Looks like geoff is building his own version of pivoted Cholesky (as an autograd function) in https://github.com/cornellius-gp/gpytorch/pull/1836, which I think is batched. Can switch over then.